### DISCUSSION TITLE
Snapshot Testing Failing

### DISCUSSION THREAD

**Discussion by automartin5000** [projen/projen#3322] [Q&A] [answered]
I'm trying a simple snapshot test on a project type that extends the JSII project type and it's failing.

Test:
```typescript
const snapshot = Testing.synth(project);
test("project snapshot", () => {
  expect(snapshot).toMatchSnapshot();
});
```
Error:
```bash
Trying to capture a snapshot of a project outside of tmpdir, which implies this test might corrupt an existing project
```

I read through [these docs](https://projen.io/docs/testing/#common-issues-and-solutions), but didn't see anything to immediately explain it. I also tried looking at projen's own tests, but didn't see any project options that looked unique other than `clobber`.

I figure it might have something to do with needing to set a temp snapshot directory? But can't figure out how to do that.

**Comment by automartin5000** [comment]
Figured it out. I did in fact need to pass it an `outdir`. I'm a bit surprised this isn't baked into the `Testing` class?

```typescript
const tmpdir = fs.mkdtempSync(path.join(os.tmpdir(), 'my-test'));
const project = new Project({ outdir: tmpdir });
```

**Reply by mrgrain** [reply]
Yes, it should be 👀

Although I didn't even know about this class. 🤣

**Comment by mrgrain** [comment, marked as answer]
This is a bug. Revert PR: https://github.com/projen/projen/pull/3334
